runtime/lua/vim/lsp/buf.lua

```diff
@@ -98,14 +96,11 @@ function M.hover(config)
     end
 
     if vim.tbl_isempty(results1) then
-      if config.silent ~= true then
```

Comment:
LMK if you think that instead we should deprecate this setting instead of just ignoring it entirely.